python/cudf/cudf/tests/test_json.py

```diff
@@ -156,6 +187,63 @@ def test_json_writer(tmpdir, pdf, gdf):
         assert_eq(pdf_string, gdf_string)
 
 
+def test_cudf_json_writer(pdf):
+    # removing datetime column because pandas doesn't support it
+    for col_name in pdf.columns:
+        if "datetime" in col_name:
+            pdf.drop(col_name, axis=1, inplace=True)
+    gdf = cudf.DataFrame.from_pandas(pdf)
+    pdf_string = pdf.to_json(orient="records", lines=True)
+    gdf_string = gdf.to_json(orient="records", lines=True, engine="cudf")
+
+    assert_eq(pdf_string, gdf_string)
+
+
+def test_cudf_json_writer_read(gdf_writer_types):
+    dtypes = {
+        col_name: col_name[len("col_") :]
+        for col_name in gdf_writer_types.columns
+    }
+    gdf_string = gdf_writer_types.to_json(
+        orient="records", lines=True, engine="cudf"
+    )
+    gdf2 = cudf.read_json(
+        StringIO(gdf_string),
+        lines=True,
+        engine="cudf_experimental",
+        dtype=dict(dtypes),
+    )
+    # zero size columns to_json() does not contain column names.
+    # So we need to add them manually
+    if len(gdf_writer_types) == 0:
+        gdf2 = cudf.DataFrame(
+            gdf2,
+            columns=gdf_writer_types.columns,
+            dtype=dict(gdf_writer_types.dtypes),
+        )
```

Comment:
Isn't this an inconsistency? Shouldn't we match with pandas' output here?

Reply:
updated to compared with pandas.
For Empty dataframe, there is a bug in pandas https://github.com/pandas-dev/pandas/issues/28558